test/f90_correct/src/io28.f90

```diff
@@ -0,0 +1,33 @@
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+! Test for I/O take derived type array reference.
+! The function should be called only one time if a function
+! reference appears in the derived type array reference.
+
+program test
```

Comment:
Is it possible to make reduce/shorten this test case ?

Reply:
These test cases have been simplified once. I don't think they need to be reduced.

Reply:
@yaoliu-hnc for clarifying.
CI still fails to build the PR, any reasons ?